ruma-events/src/lib.rs

```diff
@@ -258,6 +290,40 @@ impl RedactedSyncUnsigned {
     }
 }
 
+impl RedactedSyncUnsigned {
+    /// Convert a `RedactedSyncUnsigned` into `RedactedUnsigned`, converting the
+    /// underlying sync redaction event to a full redaction event (with room_id).
+    pub fn into_full_event(self, room_id: RoomId) -> RedactedUnsigned {
+        if let Some(why) = self.redacted_because {
+            let SyncRedactionEvent {
+                sender,
+                event_id,
+                redacts,
+                origin_server_ts,
+                unsigned,
+                content,
+                // TODO how do you want to handle this .unwrap()
```

Comment:
Could you open a separate PR that changes `redacted_because` from `Raw` to `Box`? I think that would generally be a good idea. Then we don't need any error handling here.

Reply:
This will always be a valid event, when inserting into the `redacted_because` field, so no reason to re serialize it and hold it in `Raw`.